packages/core/src/modules/connections/handlers/ConnectionRequestHandler.ts

```diff
@@ -32,16 +37,54 @@ export class ConnectionRequestHandler implements MessageHandler {
   }
 
   public async handle(messageContext: MessageHandlerInboundMessage<ConnectionRequestHandler>) {
-    const { connection, recipientKey, senderKey } = messageContext
+    const { connection, recipientKey, senderKey, message } = messageContext
 
     if (!recipientKey || !senderKey) {
       throw new AriesFrameworkError('Unable to process connection request without senderVerkey or recipientKey')
     }
 
-    const outOfBandRecord = await this.outOfBandService.findCreatedByRecipientKey(
-      messageContext.agentContext,
-      recipientKey
-    )
+    const parentThreadId = message.thread?.parentThreadId
+
+    const createOobRecord = async (did: string) => {
+      // If it's a request related to an implicit invitation, make sure destination DID is present in our wallet
+      const publicDid = tryParseDid(did)
+      if (
+        publicDid &&
+        !(await messageContext.agentContext.dependencyManager
+          .resolve(DidRepository)
+          .findCreatedDid(messageContext.agentContext, publicDid.did))
+      ) {
+        throw new AriesFrameworkError(`Referenced public did ${did} not found.`)
+      }
+
+      const outOfBandInvitation = new OutOfBandInvitation({
+        id: message.threadId,
+        label: '',
+        services: [did],
+        handshakeProtocols: [HandshakeProtocol.Connections],
+      })
+
+      const outOfBandRecord = new OutOfBandRecord({
+        role: OutOfBandRole.Sender,
+        state: OutOfBandState.AwaitResponse,
+        reusable: true,
+        autoAcceptConnection: this.connectionsModuleConfig.autoAcceptConnections,
+        isImplicitInvitation: true,
+        outOfBandInvitation,
+        tags: {
+          recipientKeyFingerprints: [recipientKey.fingerprint],
```

Comment:
Why do we store the recipientKeyFingerprints here, but not in the did exchange request handler?